lib/Dialect/TritonStructured/IR/TritonStructuredOps.cpp

```diff
@@ -21,23 +21,29 @@ void MakeTensorPtrOp::build(OpBuilder &b, OperationState &state, Value base,
                             ArrayRef<int64_t> sizes,
                             ArrayRef<OpFoldResult> strides,
                             ArrayRef<OpFoldResult> offsets,
-                            ArrayRef<OpFoldResult> parentSizes) {
-  SmallVector<int64_t> staticStrides, staticOffsets, staticParentSizes;
-  SmallVector<Value> dynamicStrides, dynamicOffsets, dynamicParentSizes;
+                            ArrayRef<OpFoldResult> shape,
+                            ArrayRef<int32_t> order) {
+  SmallVector<int64_t> staticStrides, staticOffsets, staticShape;
+  SmallVector<Value> dynamicStrides, dynamicOffsets, dynamicShape;
 
   dispatchIndexOpFoldResults(offsets, dynamicOffsets, staticOffsets);
   dispatchIndexOpFoldResults(strides, dynamicStrides, staticStrides);
-  dispatchIndexOpFoldResults(parentSizes, dynamicParentSizes,
-                             staticParentSizes);
+  dispatchIndexOpFoldResults(shape, dynamicShape, staticShape);
 
+  Type resType;
   auto basePtr = cast<triton::PointerType>(base.getType());
   auto elemType = basePtr.getPointeeType();
-  auto resType = RankedTensorType::get(sizes, basePtr);
+  if (order.empty()) {
+    resType = RankedTensorType::get(sizes, basePtr);
+  } else {
```

Comment:
nit: similar to the other comment in this file, would be good to have a comment indicating which case we're handling. Although we have a comment in the header file saying `order.empty()` means `block ptr`, but it's really not clear on first glance.